src/test/java/io/strimzi/kafka/metrics/KafkaMetricsCollectorTest.java

```diff
@@ -108,6 +109,19 @@ public void testCollectNonNumericMetric() {
         assertEquals(expectedLabels, snapshot.getDataPoints().get(0).getLabels());
     }
 
+    @Test
+    public void testLabelNameSanitizing() {
+        String labelName = PrometheusNaming.sanitizeLabelName("k-1");
+        assertEquals("k_1", labelName);
+    }
+
+    @Test
+    public void testLabelDuplicateWarning() {
```

Comment:
I think we should put this test in a new class `DataPointSnapshotBuilderTest`

Reply:
New `DataPointSnapshotBuilderTest` class created and this test is moved to there